### DISCUSSION TITLE
Help with 14. Deploy a smart contract locally using Forge

### DISCUSSION THREAD

**Discussion by pakhomovvv** [Cyfrin/foundry-full-course-cu#2466] [Q&A]
```terminal
pako@MacBookAir foundry-simple-storage-f23 % forge create SimpleStorage
[⠊] Compiling...
[⠆] Compiling 1 files with Solc 0.8.27
Error: 
solc exited with signal: 6 (SIGABRT)
dyld: Symbol not found: __ZNKSt3__115basic_stringbufIcNS_11char_traitsIcEENS_9allocatorIcEEE3strEv
  Referenced from: /Users/alexanderpakhomov/Library/Application Support/svm/0.8.27/solc-0.8.27 (which was built for Mac OS X 13.5)
  Expected in: /usr/lib/libc++.1.dylib
  ```
  Well, i can't keep going, looks like my macbook is too old (version 11.7.10)
  Can I do something? Does this affect my learning from now on? Or is it only for this video?

**Comment by EngrPips** [comment]
Read this stack-overflow https://stackoverflow.com/questions/6344560/dyld-symbol-not-found-error-how-to-resolve-this-issue post maybe it will help. Can you try uninstalling and re-installing solc.

**Reply by pakhomovvv** [reply]
no it does not work, i tried asking chatgpt, but its the same all time, the same message of error:
```terminal
MacBookAir:foundry-simple-storage-f23 pako$ forge build
[⠒] Compiling...
[⠢] Unable to resolve imports:
      "../scr/SimpleStorage.sol" in "/Users/alexanderpakhomov/foundry-test-project/foundry-f23/foundry-simple-storage-f23/script/DeploySimpleStorage.s.sol"
with remappings:
      forge-std/=/Users/alexanderpakhomov/foundry-test-project/foundry-f23/foundry-simple-storage-f23/lib/forge-std/src/
[⠒] Compiling 21 files with Solc 0.8.27
Error: 
solc exited with signal: 6 (SIGABRT)
dyld: Symbol not found: __ZNKSt3__115basic_stringbufIcNS_11char_traitsIcEENS_9allocatorIcEEE3strEv
  Referenced from: /Users/alexanderpakhomov/Library/Application Support/svm/0.8.27/solc-0.8.27 (which was built for Mac OS X 13.5)
  Expected in: /usr/lib/libc++.1.dylib
  ```

**Reply by EngrPips** [reply]
I am unsure what is wrong. I have never run into this issue before. Can you check the official Foundry repo https://github.com/foundry-rs/foundry to see if anyone has reported this same issue and how they resolved it? If no one has reported this issue, report it properly, and look forward to a response from the Foundry team.

**Reply by pakhomovvv** [reply]
I tried to change the version, bc it looks like solidity 0.8.27 isnt good for an mac OS 11, so i updated to 0.8.18, and it works well! (it was too hard to erase everything and start again, but was worth it). Huge thanks!!!

**Reply by EngrPips** [reply]
Interesting. It seems weird that `0.8.27` doesn't work, but at least you found something that works.
